requirements.txt

```diff
@@ -11,6 +11,7 @@ tqdm>=4.0
 tensorflow-metadata>=1.2.0
 betterproto<2.0.0
 packaging
+npy-append-array
```

Comment:
Not sure we want to add this as a hard requirement for Core, which would make it a hard requirement for all of Merlin. What approaches do we have to make it optional somehow?

Reply:
I made it a hard requirement because it is now a core capability of the dataset class. I think all APIs for the dataset are available in all environments. But if we want to make it optional we could break it out into a separate requirements file and have the ability to install additionally like merlin-core[np-append]...?

Reply:
It looks like the append array functionality has been merged into the mainline of NumPy as of this PR: https://github.com/numpy/numpy/pull/20321

Reply:
Resolution here is to ahead and make the package a hard dependency, since the package's functionality will be part of the NumPy 1.25 release, which alleviates concerns about ongoing maintenance.